### PR TITLE
Chain result upto compare method

### PR DESCRIPTION
**Pull request by vishrutshah**
Part of https://github.com/Azure/openapi-diff/issues/88

Verification PR : https://github.com/Azure/azure-rest-api-specs/pull/1520
Verification Build : https://travis-ci.org/Azure/azure-rest-api-specs/builds/262397614?utm_source=github_status&utm_medium=notification

TODO:
- [x] Revert the commit to remove dlls changes

### CODE REVIEW

File: lib/validate.js [hunk 1]
```diff
@@ -29,5 +29,7 @@ exports.compare = function compare(oldSwagger, newSwagger, options) {
   log.filepath = options.logFilepath || log.filepath;
   let openApiDiff = new OpenApiDiff(options);
 
-  return openApiDiff.compare(oldSwagger, newSwagger);
+  return openApiDiff.compare(oldSwagger, newSwagger).then((result, err) => {
+    console.log(result);
+  });
```
Comment (vishrutshah) [inline lib/validate.js]: .`catch` on error

File: lib/validate.js [hunk 2]
```diff
@@ -29,5 +29,7 @@ exports.compare = function compare(oldSwagger, newSwagger, options) {
   log.filepath = options.logFilepath || log.filepath;
   let openApiDiff = new OpenApiDiff(options);
 
-  return openApiDiff.compare(oldSwagger, newSwagger);
+  return openApiDiff.compare(oldSwagger, newSwagger).then((result, err) => {
```
Comment (vishrutshah) [inline lib/validate.js]: no need to `err`

File: lib/validate.js [hunk 3]
```diff
@@ -30,6 +30,10 @@ exports.compare = function compare(oldSwagger, newSwagger, options) {
   let openApiDiff = new OpenApiDiff(options);
 
   return openApiDiff.compare(oldSwagger, newSwagger).then((result, err) => {
-    console.log(result);
+    if (err) {
```
Comment (vishrutshah) [inline lib/validate.js]: no need for if (err) if we do catch

**Comment by vishrutshah** [comment]
@veronicagg & @amarzavery I've updated PR to incorporate yesterday's feedback. Please feel free to review again as you get chance. Thanks! 

Here is the successfully failed CI Build: https://travis-ci.org/Azure/azure-rest-api-specs/builds/262944817?utm_source=github_status&utm_medium=notification

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,10 @@
+## 0.1.7
+Released on 2017-08-10.
+- Chaining the promises upto `compare` method and gracefully existing application [#88](https://github.com/Azure/openapi-diff/issues/88)
```
Comment (amarzavery) [inline CHANGELOG.md]: exi~~s~~ting

File: lib/validators/openApiDiff.js [hunk 5]
```diff
@@ -61,11 +61,8 @@ class OpenApiDiff {
 
     return Promise.all([promise1, promise2]).then(results => {
       return self.processViaOpenApiDiff(results[0], results[1]).then((result, error) => {
```
Comment (amarzavery) [inline lib/validators/openApiDiff.js]:
```javascript
return Promise.all([promise1, promise2]).then(results => { 
  return self.processViaOpenApiDiff(results[0], results[1]); //this is already returning a promise, so need to chain a redundant `.then()` to it
};
```

Comment (vishrutshah) [inline lib/validators/openApiDiff.js]: 💯

**Comment by amarzavery** [comment]
:shipit:

**Comment by vishrutshah** [comment]
```nodejs
 return Promise.resolve('Thanks everyone for the review!!'); 
```

:)

**Pull request merged**